### PR TITLE
[flutter_tools] use throwToolExit

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

exit code doesn't seem to be set correctly otherwise

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/drive.dart [hunk 1]
```diff
@@ -226,7 +226,7 @@ class DriveCommand extends RunCommandBase {
       await driverService.stop(userIdentifier: userIdentifier, writeSkslOnExit: skslFile);
     }
     if (testResult != 0) {
-      return FlutterCommandResult.fail();
+      throwToolExit();
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/commands/drive.dart:229]: We need a test for this. Interestingly, we already have a test for this: https://github.com/flutter/flutter/blob/master/packages/flutter_driver/test_driver/failure_test.dart. Not sure how it didn't catch the issue.

**Comment by jonahwilliams** [comment]
Need to land this ASAP to prevent any more breakages from creeping in

File: packages/flutter_tools/lib/src/commands/drive.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/drive.dart]: It must not be running, I'll look through the test configs

**Review by yjbanov — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/drive.dart (hunk 1 above)
Comment (yjbanov) [reply packages/flutter_tools/lib/src/commands/drive.dart]: It is running, but it seems to be failing for an unrelated reason and masking our failure to report a test failure:

```
AndroidManifest.xml could not be found.
Please check /home/yjbanov/code/flutter/flutter/packages/flutter_driver/android/AndroidManifest.xml for errors.
[!] The build process for Android has changed, and the current project configuration is no longer valid. Please consult

https://github.com/flutter/flutter/wiki/Upgrading-Flutter-projects-to-build-with-gradle

for details on how to upgrade the project.
```

To fix this, we need to add a success test next to the failure test in the same package.

**Comment by yjbanov** [comment]
> Need to land this ASAP to prevent any more breakages from creeping in

SGTM

**Pull request merged**